bkarachok/lab1-BasicIO/user.c

```diff
@@ -0,0 +1,198 @@
+/******************************************************************************/
+/* Files to Include                                                           */
+/******************************************************************************/
+
+#ifdef __XC32
+#include <xc.h>          /* Defines special function registers, CP0 regs  */
+#endif
+
+#include <stdint.h>          /* For uint32_t definition                       */
+#include <stdbool.h>         /* For true/false definition                     */
+#include "user.h"            /* variables/params used by user.c               */
+
+/******************************************************************************/
+/* User Functions                                                             */
+
+/******************************************************************************/
+
+void InitApp(void) {
+    /* Setup analog functionality and port direction */
+
+    // LED output
+    // Disable analog mode for G6
+    ANSELGbits.ANSG6 = 0;
+    // Set direction to output for G6
+    TRISGbits.TRISG6 = 0;
+    
+    // Initialize outputs for other LEDs
+    
+    // D4 hasn't analog mode
+    // Set direction to output for D4
+    TRISDbits.TRISD4 = 0;
+    
+    // Disable analog mode for B11
+    ANSELBbits.ANSB11 = 0;
+    // Set direction to output for B11
+    TRISBbits.TRISB11 = 0;
+    
+    // Disable analog mode for G15
+    ANSELGbits.ANSG15 = 0;
+    // Set direction to output for G15
+    TRISGbits.TRISG15 = 0;
+ 
+    // Turn on LEDs for testing
+    LD1_PORT_BIT = 0;
+    LD2_PORT_BIT = 0;
+    LD3_PORT_BIT = 0;
+    LD4_PORT_BIT = 0;
+    // Stop program here
+    
+    //while(1); 
+    
+    // BTN1 input
+    // Disable analog mode
+    ANSELAbits.ANSA5 = 0;
+    // Set directions to input
+    TRISAbits.TRISA5 = 1;
+
+    // Initialize input for BTN2
+    
+    TRISAbits.TRISA4 = 1;
+    
+    // Test switches
+    
+   /* while (1) {
+        LATGbits.LATG6  = PORTAbits.RA5;
+        LATDbits.LATD4  = PORTAbits.RA4;
+    };
+   */   
+}
+
+void Delay(int n) {
+    volatile int i;
+    for (i = 0; i < n; i++) {
+    }
+}
+
+void Flash_LED(void) {
+    int delay_count;
+    while (1) {
```

Comment:
Generally infinite cycles should be placed outside of functions, somewhere in `main`